smt/surrogate_models/krg_based.py

```diff
@@ -1159,7 +1159,7 @@ def _predict_values(self, x: np.ndarray, is_acting=None) -> np.ndarray:
             Evaluation point output variable values
         """
         # Initialization
-        if is_acting is None:
+        if is_acting is not None or not (self.design_space.is_all_cont):
```

Comment:
This is weird because you never use the `is_acting` array content given as argument.

Reply:
It is used by gower_componentwise_distances

Reply:
I mean you have changed the logic. Before your fix when `is_acting` was None  we retrieve it from ` self.design_space.correct_get_acting(x)`, otherwise we just use it.

Now if we pass an `is_acting` array, you seem to override it with `x, is_acting = self.design_space.correct_get_acting(x)`. So whether or not it is right, the intent is not clear and it looks a bit confusing.

Moreover as all the tests were successful before I think we should add some tests to cover the logic we want and document the expected behaviour as well.